lib/src/transformer/declaration_parsing.dart

```diff
@@ -56,8 +56,18 @@ class ParsedDeclarations {
     unit.declarations.forEach((CompilationUnitMember member) {
       member.metadata.forEach((annotation) {
         var name = annotation.name.toString();
+        var publicMember;
 
-        declarationMap[name]?.add(member);
+        if (member is ClassDeclaration && member.name.name.startsWith('_\$')) {
+          unit.declarations.forEach((CompilationUnitMember innerLoopMember) {
```

Comment:
We'll probably want some validation to make sure that there is only 1 instance of a public props class in the file. We could do:
```suggestion
          var matchingPublicPropsOrStateClasses = 
          	unit.declarations.where((CompilationUnitMember innerLoopMember) => 
          	innerLoopMember is ClassDeclaration && 
				member.name.name.substring(2) == innerLoopMember.name.name);
```	

Then ensure that the length of `matchingPUblicPropsOrStateClasses` is 1 and assign it to `publicMember`. If the length is not 1, we should tell consumers that they should not write their own public props classes (or delete the one that is there), and instead rely on the code modded ones.

Reply:
Also, looks like this code runs on all annotations, we should only search for the public props/state class if the annotation is one of `@Props()`, `@State()`, `@AbstractProps()`, `@AbstractState()`